pkg/operator/resources/resources.go

```diff
@@ -147,24 +156,45 @@ func DeleteAPI(apiName string, keepCache bool) (*schema.DeleteResponse, error) {
 		return nil, err
 	}
 
-	if deployedResource == nil {
```

Comment:
Based on the conversation, the "just to be sure" deletion can be brought back. Also add api splitter to the just to be sure deletion.

Reply:
I added it back already.

Reply:
The nil check should happen before `if deployedResource.Kind == userconfig.SyncAPIKind {` is called because you may run into a nil pointer exception if deployedResource is nil.